src/graph.py

```diff
@@ -47,3 +47,29 @@ def get_spanning_tree(self, start):
 
         return spanning_tree
 
+    def get_spanning_tree(self, start, seed):
+        random.seed(seed)
+        spanning_tree = Graph(self.num_nodes)
+        stack = []
+        visited = [False]*self.num_nodes
+
+        visited[start] = True
+        stack.append(start)
+        while stack:
+            node = stack[len(stack)-1]
+            unvisited_nodes = []
+            for next_node in self.graph[node]:
+                if not visited[next_node]:
+                    unvisited_nodes.append(next_node)
+            if unvisited_nodes:
+                next_node = random.choice(unvisited_nodes)
+                stack.append(next_node)
+                visited[next_node] = True
+                spanning_tree.add_edge(node, next_node)
+                spanning_tree.add_edge(next_node, node)
+            else:
+                stack.pop()
+
+        return spanning_tree
+
+
```

Comment:
We should see if there could be a way to do this without copying and pasting code

Reply:
Done